api/drivers/qemu/driver.js

```diff
@@ -223,14 +257,38 @@ class QemuDriver extends Driver {
    * @return {Promise[Machine]}
    */
   refresh(machine) {
-    let requestOptions = {
-      url: 'http://' + machine.ip + ':' + machine.plazaport,
-      method: 'GET'
-    };
-    return request(requestOptions)
-      .then(() => {
-        machine.status = 'running';
-        return machine;
+    return ConfigService.get('qemuServiceURL', 'qemuServicePort')
+      .then((config) => {
+        if (machine.status === 'booting') {
```

Comment:
`refresh` should not guess on machine's status because part of its job is to actually refresh the machine's status.